llvm/lib/Target/NVPTX/NVPTXIncreaseAlignment.cpp

```diff
@@ -0,0 +1,131 @@
+//===-- NVPTXIncreaseAlignment.cpp - Increase alignment for local arrays --===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// A simple pass that looks at local memory arrays that are statically
+// sized and sets an appropriate alignment for them. This enables vectorization
```

Comment:
Nit: So, does the pass sets the alignment or *increases* it? I think it's the former, as 'increase' implies that it's already set, which is not necessarily the case. We may keep the wording consistent between the pass/file names and the comments/ and what the code does.

Something along the lines of "NVPTXSetDefaultLocalAlignment" might work.

Reply:
SGTM.